lib/school.rb

```diff
@@ -0,0 +1,46 @@
+class School
+
+  attr_reader :start_time,
+              :total_hours,
+              :name
+
+  def initialize(start_time, total_hours)
+    @start_time = start_time
+    @total_hours = total_hours
+    @names =[]
+  end
+
+  def hours_in_school_day()
+    hours_in_school_day = @total_hours
+  end
+
+  def add_student_name(name)
+   @names.push(name)
+  end
+
+  def student_names
+    student_names = @names
+  end
+
+  def end_time()
+    time = self.start_time.to_i + self.total_hours
+    "#{time}:00"
+  end
+
+  def is_full_time?
+    self.total_hours >=4
+  end
+
+  def standard_student_names()
```

Comment:
This works which is a great first step! One thing is I would avoid using the `!` entirely to mutate values in Ruby. It is generally a bad practice as it causes some unintended side effects.

Also here you are using `standard_names` but never actually returning the value from this method.

So remove the `!` and just return `standard_names` after your each loop would be the fixes I would make here